docs/guide/migration.md

```diff
@@ -101,6 +87,15 @@ You can use `?init` which is similar to the previous behavior.
 })
 ```
 
+## Experimental
+
+### Using esbuild deps optimization at build time
+
+In v3, Vite allows the use of esbuild to optimize dependencies during build time. If enabled, it removes one of the most significant differences between dev and prod present in v2. [`@rollupjs/plugin-commonjs`](https://github.com/rollup/plugins/tree/master/packages/commonjs) is no longer needed in this case since esbuild converts CJS-only dependencies to ESM.
+
+If you want to try this build strategy, you can use `optimizeDeps.disabled: false` (the default in v3 is `disabled: 'build'`). `@rollup/plugin-commonjs`
+can be removed by passing `build.commonjsOptions: { include: [] }`
```

Comment:
I think we should document this somewhere else as users don't have to read this when migrating. It doesn't look like we have a `experimental-options.md`, but maybe it make sense to include this in the prebundling guide, which could also be worth updating later as a whole https://github.com/vitejs/vite/issues/8735